cmd/deployIntegrationArtifact.go

```diff
@@ -0,0 +1,158 @@
+	// Utils can be used wherever the command.ExecRunner interface is expected.
+	// It can also be used for example as a mavenExecRunner.
+	utils := newDeployIntegrationArtifactUtils()
+	utils.Stdout(log.Writer())
+	httpClient := &piperhttp.Client{}
+	// For HTTP calls import  piperhttp "github.com/SAP/jenkins-library/pkg/http"
+	// and use a  &piperhttp.Client{} in a custom system
+	// Example: step checkmarxExecuteScan.go
+
+	// Error situations should be bubbled up until they reach the line below which will then stop execution
+	// through the log.Entry().Fatal() call leading to an os.Exit(1) in the end.
+	err := runDeployIntegrationArtifact(&config, telemetryData, httpClient)
+	if err != nil {
+		log.Entry().WithError(err).Fatal("step execution failed")
+	}
+}
+
+func runDeployIntegrationArtifact(config *deployIntegrationArtifactOptions, telemetryData *telemetry.CustomData, httpClient piperhttp.Sender) error {
+	cookieJar, cookieErr := cookiejar.New(nil)
+	if cookieErr != nil {
+		return errors.Wrap(cookieErr, "creating a cookie jar failed")
+	}
+	clientOptions := piperhttp.ClientOptions{
+		CookieJar: cookieJar,
+	}
+	httpClient.SetOptions(clientOptions)
+
+	header := make(http.Header)
+	header.Add("Accept", "application/json")
+
+	deployURL := config.Host +
+		"/api/v1/DeployIntegrationDesigntimeArtifact?Id=" + "'" + config.IntegrationFlowID +
+		"'" + "&Version=" + "'" + config.IntegrationFlowVersion + "'"
+
+	finalResult, err := getBearerTokenforDeployIntegrationArtifactCall(config, httpClient)
+	if err != nil {
+		return errors.Wrap(err, "failed to fetch Bearer Token")
+	}
+	clientOptions.Token = "Bearer " + finalResult
+	httpClient.SetOptions(clientOptions)
+
+	deployResp, httpErr := httpClient.SendRequest("POST", deployURL, nil, header, nil)
+
+	defer func() {
+		if deployResp != nil && deployResp.Body != nil {
+			deployResp.Body.Close()
+		}
+	}()
+
+	if deployResp == nil {
+		return errors.Errorf("did not retrieve a HTTP response: %v", httpErr)
+	}
+
+	if deployResp.StatusCode == 202 {
+		log.Entry().
+			WithField("IntegrationFlowID", config.IntegrationFlowID).
+			Info("successfully deployed in to CPI runtime")
+		return nil
+	}
+
+	log.Entry().Errorf("a HTTP error occurred! Response Status Code: %v", deployResp.StatusCode)
+	return errors.Wrap(httpErr, "Deploying the integration flow failed")
+}
+
+func getBearerTokenforDeployIntegrationArtifactCall(config *deployIntegrationArtifactOptions, httpClient piperhttp.Sender) (string, error) {
+
+	cookieJar, cookieErr := cookiejar.New(nil)
+	if cookieErr != nil {
+		return "", errors.Wrap(cookieErr, "creating a cookie jar failed")
+	}
+	clientOptions := piperhttp.ClientOptions{
+		CookieJar: cookieJar,
+		Username:  config.Username,
+		Password:  config.Password,
+	}
+	httpClient.SetOptions(clientOptions)
+
+	header := make(http.Header)
+	header.Add("Accept", "application/json")
+	tokenURL := config.OAuthTokenProviderURL + "?grant_type=client_credentials"
```

Comment:
Maybe it is prettier with `fmt.Sprintf()`, as in:
```suggestion
	tokenURL := fmt.Sprintf("%s?grant_type=client_credentials", config.OAuthTokenProviderURL)
```